### ISSUE TITLE
Bug Version 2.76.0 azurerm_kubernetes_cluster_node_pool adds by default a os_sku = "Ubuntu" when I deploy a linux node and it fails

### ISSUE THREAD

**Issue by mrlunchbox777** [hashicorp/terraform-provider-azurerm#13312]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Expected Behavior

A minor version update on AzureRM provider should give me a no-op upgrade, or at least provide a path to do so.

### Actual Behavior

A minor version tried to insert a `os_sku = "Ubuntu"` on the cluster with no changes to the terraform, and was trying to force a replacement on the apply

### Steps to Reproduce

1. Create a cluster using AzureRM 2.75.0
2. Upgrade to AzureRM 2.76.0
3. Notice that terraform wants to replace your cluster and there is no way to no-op this.....

### References

* This is closely related to #13306, but with a Linux node instead of a Windows one

**Comment by mrlunchbox777** [comment]
I believe this PR had code that caused this - https://github.com/hashicorp/terraform-provider-azurerm/pull/13284/commits/5f758ce8590d8c3ffcf303ada44269e122f625d8

**Comment by arogic** [comment]
Hi, I encounter the same issue

```
[error]Terraform command 'apply' failed with exit code '1'.:  creating/updating Managed Kubernetes Cluster Node Pool "win" (Resource Group "<rg-name>"): containerservice.AgentPoolsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidOSSKU" Message="OSSKU='Ubuntu' is invalid, details: Windows does not allow OSSKU selection"
```
I have not specified the os_sku in my terraform resource "azurerm_kubernetes_cluster_node_pool" using os_type = "Windows". 

```
resource "azurerm_kubernetes_cluster_node_pool" "node_pool" {
  name                  = var.node_pool.name
  kubernetes_cluster_id = azurerm_kubernetes_cluster.cluster.id
  vm_size               = var.node_pool.vm_size
  node_count            = var.node_pool.node_count
  os_disk_size_gb       = var.node_pool.os_disk_size_gb
  os_type               = var.node_pool.os_type
  tags                  = var.tags
}
```
Thanks in advance

Regards

**Comment by ms-henglu** [comment]
Thanks for reporting this bug. I'll fix it asap. Sorry for it. 🙏 

The cause is os_sku defaults to `Ubuntu` which is not appliable when using Windows.

**Comment by smark88** [comment]
additionally if you are using ubuntu it forces a cluster rebuild https://github.com/hashicorp/terraform-provider-azurerm/issues/13306#issuecomment-919223483

So far I found i can 
```
resource "azurerm_kubernetes_cluster" "main" {
...
  lifecycle {
    ignore_changes = [default_node_pool.0.os_sku]
  }
}
```
same for the nodes
```
resource "azurerm_kubernetes_cluster_node_pool" "main" {
...
  lifecycle {
    ignore_changes = [os_sku]
  }
}
```

**Comment by aristosvo** [comment]
Seems fixed by #13222

**Comment by aristosvo** [comment]
Seems fixed by #13321, which is released in 2.77.0

**Comment by mrlunchbox777** [comment]
yes, @aristosvo this was fixed as far as I can tell in 2.77.0. Thanks so much!
